Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -0,0 +1,213 @@
+//! current supported compression formats
+std::vector<std::string> supportedCompressionFormats{"gz"};
+
+//! determines whether file and compression format are valid
+bool validate(std::string& fileFormat, std::string& compressionFormat) {
+  bool flag_fileFormat = true;
+  bool flag_compressionFormat = true;
+
+  //! Case 1: No compression format
+  if (compressionFormat.empty()) {
+    //! Unconventional format types
+    if (fileFormat.compare("maegz") == 0) {
+      fileFormat = "mae";
+      compressionFormat = "gz";
+      return true;
+    }
+
+    //! Usual file formats
+    return std::find(supportedFileFormats.begin(), supportedFileFormats.end(),
+                     fileFormat) != supportedFileFormats.end();
+  }
+  //! Case 2: With compression format
+  else {
+    flag_fileFormat =
+        std::find(supportedFileFormats.begin(), supportedFileFormats.end(),
+                  fileFormat) != supportedFileFormats.end();
+    flag_compressionFormat =
+        std::find(supportedCompressionFormats.begin(),
+                  supportedCompressionFormats.end(),
+                  compressionFormat) != supportedCompressionFormats.end();
+
+    //! Case A: Valid file format and valid compression format
+    if (flag_fileFormat && flag_compressionFormat) {
+      return true;
+    }
+
+    //! Case B, C: Valid file format but not valid compression format
+    //! 						or invalid file format
+    //! and valid compression format
+    else if ((flag_fileFormat && !flag_compressionFormat) ||
+             (!flag_fileFormat && flag_compressionFormat)) {
+      return false;
+    }
+
+    //! Case D: Invalid file format and invalid compression format
+    else {
+      //! it is possible that we have read a file with name *.2.txt (for
+      //! example) thus fileformat = "2" and compressionformat = "txt", so we
+      //! try to set the file format as the compression format and the
+      //! compression format as an empty string. Then we check of validity.
+      fileFormat = compressionFormat;
+      compressionFormat = "";
+      return std::find(supportedFileFormats.begin(), supportedFileFormats.end(),
+                       fileFormat) != supportedFileFormats.end();
+    }
+  }
+}
+
+//! returns the file name givens the file path
+std::string getFileName(const std::string path) {
+  char delimiter = '/';
+  std::string fname = "";
+  auto slash = path.rfind(delimiter, path.length());
+  if (slash != std::string::npos) {
+    fname += path.substr(slash + 1, path.length() - slash);
+  }
+  return fname;
+}
+
+//! given file path determines the file and compression format
+void determineFormat(const std::string path, std::string& fileFormat,
+                     std::string& compressionFormat) {
+  std::string fileName = getFileName(path);
+  int dots = std::count(fileName.begin(), fileName.end(), '.');
+
+  if (dots == 0) {
+    throw std::invalid_argument(
+        "Recieved Invalid File Format, no extension or compression");
```

Comment:
It isn't so much as "invalid" but "unsupported" or "unknown". What about "Unable to determine file format: no filename extension found" or something like that?